File: temp.py
from flask import Flask, request, render_template

app = Flask(__name__)

@app.route('/')
def temperature_converter():
    return render_template('index.html')

@app.route('/convert', methods=['POST'])
def convert_temperature():
    if 'celsius' in request.form:
        try:
            celsius = float(request.form['celsius'])
            fahrenheit = celsius * 9/5 + 32
            return render_template('result.html', celsius=celsius, fahrenheit=fahrenheit)
        except ValueError:
            return render_template('error.html', message='Invalid input')
    elif 'fahrenheit' in request.form:
        try:
            fahrenheit = float(request.form['fahrenheit'])
            celsius = (fahrenheit - 32) * 5/9
            return render_template('result.html', celsius=celsius, fahrenheit=fahrenheit)
        except ValueError:
            return render_template('error.html', message='Invalid input')
    else:
        return render_template('error.html', message='No input provided')

if __name__ == '__main__':
    app.run()

File: temp.php
<?php

class HomeController
{
    public function index()
    {
        return $this->view();
    }

    public function convertTemperature($form)
    {
        if (isset($form['celsius'])) {
            try {
                $celsius = floatval($form['celsius']);
                $fahrenheit = $celsius * 9 / 5 + 32;
                $data['Celsius'] = $celsius;
                $data['Fahrenheit'] = $fahrenheit;
                return $this->view('Result', $data);
            } catch (Exception $e) {
                $data['ErrorMessage'] = 'Invalid input';
                return $this->view('Error', $data);
            }
        } elseif (isset($form['fahrenheit'])) {
            try {
                $fahrenheit = floatval($form['fahrenheit']);
                $celsius = ($fahrenheit - 32) * 5 / 9;
                $data['Celsius'] = $celsius;
                $data['Fahrenheit'] = $fahrenheit;
                return $this->view('Result', $data);
            } catch (Exception $e) {
                $data['ErrorMessage'] = 'Invalid input';
                return $this->view('Error', $data);
            }
        } else {
            $data['ErrorMessage'] = 'No input provided';
            return $this->view('Error', $data);
        }
    }

    private function view($viewName = '', $data = [])
    {
        // Implement your view rendering logic here
        // Return the rendered view
    }
}
